### PR TITLE
Fix signaltools `medfilt` for `scipy>=1.7.0`

### PR DESCRIPTION
**Pull request by emcastillo**
This fixes the actual routine while
#5368 deals with the tests

Ref: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.medfilt.html

### CODE REVIEW

**Comment by emcastillo** [comment]
rtd failure seems unrelated

File: cupyx/scipy/signal/signaltools.py [hunk 1]
```diff
@@ -509,18 +509,17 @@ def medfilt(volume, kernel_size=None):
     .. seealso:: :func:`cupyx.scipy.ndimage.median_filter`
     .. seealso:: :func:`scipy.signal.medfilt`
     """
-    if volume.dtype.kind == 'c':
-        # scipy doesn't support complex
-        raise ValueError("complex types not supported")
-    # output is forced to float64 to match scipy
+    if volume.dtype not in (cupy.uint8, cupy.float32, cupy.float64):
+        # Scipy's version only supports uint8, float32, and float64
+        raise ValueError("only supports uint8, float32, and float64")
```
Comment (toslunar) [inline cupyx/scipy/signal/signaltools.py:514]: The old check seems correct.
```
>>> scipy.signal.medfilt(np.array([3, 1, 4, 1, 5, 9, 2, 6], np.int16))
array([1, 3, 1, 4, 5, 5, 6, 2], dtype=int16)
>>> scipy.__version__
'1.7.0'
```

**Review by toslunar — CHANGES_REQUESTED** [review]

File: cupyx/scipy/signal/signaltools.py (hunk 1 above)
Comment (emcastillo) [reply cupyx/scipy/signal/signaltools.py]: good catch! thanks :)

**Review by toslunar — COMMENTED** [review]
Could you revert the order of `for_all_dtypes` and `numpy_cupy_allclose`?
```diff
--- a/tests/cupyx_tests/scipy_tests/signal_tests/test_signaltools.py
+++ b/tests/cupyx_tests/scipy_tests/signal_tests/test_signaltools.py
@@ -255,9 +255,9 @@ class TestOrderFilter(unittest.TestCase):
 @testing.gpu
 @testing.with_requires('scipy>=1.7.0')
 class TestMedFilt(unittest.TestCase):
+    @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(atol=1e-8, rtol=1e-8, scipy_name='scp',
                                  accept_error=ValueError)  # for even kernels
-    @testing.for_all_dtypes()
     def test_medfilt(self, xp, scp, dtype):
         if sys.platform == 'win32':
             pytest.xfail('medfilt broken for Scipy 1.7.0 in windows')
@@ -275,9 +275,9 @@ class TestMedFilt(unittest.TestCase):
 @testing.gpu
 @testing.with_requires('scipy>=1.7.0')
 class TestMedFilt2d(unittest.TestCase):
+    @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(atol=1e-8, rtol=1e-8, scipy_name='scp',
                                  accept_error=ValueError)  # for even kernels
-    @testing.for_all_dtypes()
     def test_medfilt2d(self, xp, scp, dtype):
         if sys.platform == 'win32':
             pytest.xfail('medfilt2d broken for Scipy 1.7.0 in windows')
```

**Comment by emcastillo** [comment]
done!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Review by toslunar — APPROVED** [review]

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Thanks @toslunar :pray:

**Comment by toslunar** [comment]
I'll backport dad44ee7b7bbd410882ea0a9e6c48d5e6e6f6c99.

**Comment by toslunar** [comment]
I suppose this PR should be backported.  The corrected tests (#5518) are failing in v9.

**Pull request merged**